codesigning/signing-worker/natcap-codesign.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+"""Service script to sign InVEST windows binaries."""
+
+import json
+import logging
+import os
+import shutil
+import subprocess
+import sys
+import textwrap
+import time
+
+import pexpect  # apt install python3-pexpect
+import requests  # apt install python3-requests
+
+LOGGER = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+CERTIFICATE = sys.argv[1]
+
+FILE_DIR = os.path.dirname(__file__)
+TOKEN_FILE = os.path.join(FILE_DIR, "access_token.txt")
+with open(TOKEN_FILE) as token_file:
+    ACCESS_TOKEN = token_file.read().strip()
+
+
+def get_from_queue():
+    """Get an item to sign from the queue.
+
+    Returns:
+        ``None`` if there are no items in the queue, the JSON response dict
+        otherwise.
+    """
+    response = requests.post(
+        "https://us-west1-natcap-servers.cloudfunctions.net/codesigning-queue",
+        headers={"Content-Type": "application/json"},
+        json={
+            "token": ACCESS_TOKEN,
+            "action": "dequeue"
+        })
+    if response.status_code == 204:
+        return None
+    else:
+        return response.json()
+
+
+def download_file(url):
+    """Download an arbitrarily large file.
+
+    Adapted from https://stackoverflow.com/a/16696317
+
+    Args:
+        url (str): The URL to download.
+
+    Returns:
+        ``None``
+    """
+    local_filename = url.split('/')[-1]
+    with requests.get(url, stream=True) as r:
+        r.raise_for_status()
+        with open(local_filename, 'wb') as f:
+            for chunk in r.iter_content(chunk_size=8192):
+                f.write(chunk)
+    return local_filename
+
+
+def upload_to_bucket(filename, path_on_bucket):
+    """Upload a file to a GCS bucket.
+
+    Args:
+        filename (str): The local file to upload.
+        path_on_bucket (str): The path to the file on the GCS bucket, including
+            the ``gs://`` prefix.
+
+    Returns:
+        ``None``
+    """
+    subprocess.run(['gsutil', 'cp', filename, path_on_bucket], check=True)
```

Comment:
How does the signing worker authenticate to `gsutil`?

Reply:
There's a line in the Ansible playbook that does the authentication, so the service script pretty much "just works"